YellowNotes.Api/Controllers/CategoriesController.cs

```diff
@@ -0,0 +1,77 @@
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using YellowNotes.Api.Extensions;
+using YellowNotes.Core.Dtos;
+using YellowNotes.Core.Repositories;
+
+namespace YellowNotes.Api.Controllers
+{
+    [ApiController]
+    [Route("/[controller]")]
+    public class CategoriesController : ControllerBase
+    {
+        private readonly ICategoryRepository categoryRepository;
+
+        public CategoriesController(ICategoryRepository categoryRepository) =>
+            this.categoryRepository = categoryRepository;
+
+        [HttpGet]
+        [ProducesResponseType(typeof(IEnumerable<CategoryDto>), StatusCodes.Status200OK)]
+        public async Task<ActionResult<IEnumerable<CategoryDto>>> GetCategories(
+            CancellationToken cancellationToken = default)
+        {
+            var userEmail = HttpContext.GetEmailFromClaims();
+            var categories = await categoryRepository.GetCategories(userEmail, cancellationToken);
+            return Ok(categories);
+        }
+
+        [HttpPost]
+        [ProducesResponseType(typeof(CategoryDto), StatusCodes.Status200OK)]
+        [ProducesResponseType(StatusCodes.Status400BadRequest)]
+        [ProducesResponseType(StatusCodes.Status422UnprocessableEntity)]
+        public async Task<ActionResult> CreateCategory([FromBody] CategoryDto category,
+            CancellationToken cancellationToken = default)
+        {
+            if (category.CategoryId != 0)
+            {
+                return BadRequest();
+            }
+
+            var userEmail = HttpContext.GetEmailFromClaims();
+            var result = await categoryRepository
+                .CreateCategory(category, userEmail, cancellationToken);
+
+            if (result == null)
+            {
+                return UnprocessableEntity("Failed to create category");
+            }
+            return Ok(result);
```

Comment:
good practice is to return Created with a url to newly created resource instead of Ok.

Reply:
It's pretty redundant in this case because we won't be accessing individual categories.